### PR TITLE
Sovereign Cloud Support

### PR DESCRIPTION
**Pull request by jasoncoolmax**
Add support for the instance aware flow.
(Additional fix in common core needed is here: https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/207)

### CODE REVIEW

File: MSAL/src/requests/MSALInteractiveRequest.m [hunk 1]
```diff
@@ -151,7 +151,12 @@ - (void)acquireTokenImpl:(MSALCompletionBlock)completionBlock
             {
                 _code = oauthResponse.authorizationCode;
 
-                // TODO: handle MSIDWebOAuth2Response and instance aware flow (cloud host)
+                // handle instance aware flow (cloud host)
+                if ([response isKindOfClass:MSIDWebAADAuthResponse.class])
+                {
+                    MSIDWebAADAuthResponse *aadResponse = (MSIDWebAADAuthResponse *)response;
+                    [_parameters setCloudAuthorityWithCloudHostName:aadResponse.cloudHostName];
```
Comment (oldalton) [inline MSAL/src/requests/MSALInteractiveRequest.m]: do we need to save the whole cloudAuthority in parameters if we only use host from it?

File: MSAL/test/unit/MSALAcquireTokenTests.m [hunk 2]
```diff
@@ -214,12 +218,104 @@ - (void)testAcquireTokenSilent_whenNoATForScopeInCache_shouldUseRTAndReturnNewAT
...
+                             authority:nil
+                         correlationId:nil
+                       completionBlock:^(MSALResult *result, NSError *error)
+     {
+         XCTAssertNil(error);
+         XCTAssertNotNil(result);
+         XCTAssertEqualObjects(result.accessToken, @"i am an updated access token!");
+         
+         // Expect authority to be cloud authority
+         XCTAssertEqualObjects(result.authority, @"https://login.microsoftonline.de/" DEFAULT_TEST_UTID);
```
Comment (oldalton) [inline MSAL/test/unit/MSALAcquireTokenTests.m]: can we also verify that the token was cached under correct authority?

File: MSAL/src/requests/MSALInteractiveRequest.m (hunk 1 above)
Comment (jasoncoolmax) [reply MSAL/src/requests/MSALInteractiveRequest.m]: In `[MSALRequestParameters msidConfiguration]` it needs the whole `cloudAuthority` :)

File: MSAL/test/unit/MSALAcquireTokenTests.m (hunk 2 above)
Comment (jasoncoolmax) [reply MSAL/test/unit/MSALAcquireTokenTests.m]: agree, done :)

**Pull request merged**